enterprise/views.py

```diff
@@ -650,16 +650,23 @@ def get_base_details(self, enterprise_uuid, course_run_id):
         """
         try:
             course, course_run = CourseCatalogApiServiceClient().get_course_and_course_run(course_run_id)
-        except ImproperlyConfigured:
+        except ImproperlyConfigured as error:
```

Comment:
So I don't think prod has an improper configuration for the course catalog API, so this part might not be necessary, otherwise it would happen for every single course.

Reply:
So what do u think, only raising 404 in `ImproperlyConfigured` would be sufficient if this exception raised due to any reason in future?

Reply:
This is only raised I believe if in the LMS Django admin we haven't set up a catalog integration. But if that's the case, it would *always* happen, on every single call.

Reply:
If the call fails somewhere inside of `get_course_and_course_run`, we already log the error for that in `_load_data`.

Reply:
tl;dr this log is actually unnecessary and the traceback would give more than enough information (it would explicitly say "Catalog Integration does not exist" or something).